### ISSUE TITLE
Hot to theme QuickSwitch border size and colour?

### ISSUE THREAD

**Issue by fladd** [bbidulock/icewm#736]
I cannot seem to find a way to change the border size and colour of the QuickSwitch window. It does not follow the normal window settings.

**Comment by gijsbers** [comment]
Does `grep -e Color.*Quick -e Quick.*Marg ~/.icewm/preferences` help?

**Comment by fladd** [comment]
Unfortunately not. It gives me this:
<img width="737" alt="image" src="https://github.com/bbidulock/icewm/assets/2971539/06904244-e530-4729-ad99-ffb539cb29f7">

Nothing on borders I am afraid.

**Comment by gijsbers** [comment]
OK, it's just a generic function with the color derived from the generic switch background:
```
    g.setColor(switchBg);
    g.drawBorderW(0, 0, width() - 1, height() - 1, true);
```

**Comment by fladd** [comment]
Not only this, but it is also always a 3D style, even when Look=flat.

It would be great if this could be fully themed so I could make it look
consistent with how my windows and my menu looks.

**Comment by gijsbers** [comment]
Maybe other looks need a fix as well. `fully themed` is undefined and could be a lot of unattractive work with unclear benefits. The QuickSwitch already has 12 themables and 13 preferences. Thanks for this observation.

**Comment by fladd** [comment]
Well, "fully" might have been not the best choice of word. I meant that it would be great if it were consistent with what can be styled on the other elemens (e.g. windows/menu), such that one can create themes that look coherent.

**Comment by Code7R** [comment]
I am not sure that his is an enhancement only. For certain themes, even the included ones, quick-switch looks outright ugly (low contrast, hard to read). Example: NanoBlue, although the actual cause is the color configured in the theme - when the background pixmap is not used, the raw color is simply bad.

@fladd Would it make sense to simply mimic the menu look in the quickswitch overlay? I.e. matching Look and color scheme completely?

**Comment by fladd** [comment]
@Code7R Yes, that would make sense, but keep in mind that the menu look is actually affected by several things that interact in a complex (and not very intuitive) way. For instance, to get a 1px border in colour X around the menu, one would need to set "ColorActiveMenuItem" to X while at the same time have "ColorNormalMenu" set and have a "menubg.xpm" file present.

In this context it is also worth noting that the colour you set in "ColorNormalMenu" will actually not be put on screen as is, but a significantly lighter version of it will appear on screen. So for instance, if you want it to appear on screen as #C4C3C8, you need to set it to #939297 for some reason.

So yes, it would be nice if the QuickSwitch would follow the menu, but it would then also need to mirror all these complex oddities :-)

**Comment by gijsbers** [comment]
https://github.com/ice-wm/icewm/issues/174#issuecomment-2717323112
